### ISSUE TITLE
Windows setup?

### ISSUE THREAD

**Issue by tombrown08** [alexa/interactive-adventure-game-tool#21]
does anyone have instructions on how to do this in a windows environment rather than mac or is it not possible?

**Comment by ginareams** [comment]
I'm looking for the same answer. I was under the impression (including from the website), until the final steps in the ReadMe, that this was possible on Windows.

**Comment by ferguson4848** [comment]
Will this not run on Windows?

**Comment by AlBakerKY** [comment]
I successfully ran this on windows. It shouldn't deviate too much, BUT see the answer from @MureedSultan  on https://github.com/alexa/interactive-adventure-game-tool/issues/3.  I hit that issue.

**Comment by ferguson4848** [comment]
Thanks for pointing me in that direction. But where did you run those
commands from? I see in the Readme that this is run using Node.js and with
a standard web browser. But I'm a bit confused on how to get that function
on Windows?

I'd really appreciate any guidance.

Thanks

**Comment by AlBakerKY** [comment]
That (from @MureedSultan) is a change to upload.js.  

The installation commands themselves are run from the command line. 'Brew' is os x centric, so you can skip that, but you will need to install node (I have v6.8.0).  NPM comes with it.

**Comment by ferguson4848** [comment]
Okay, so I did install Node, but how do I create a file under a ".aws"
directory?

**Comment by ginareams** [comment]
Hey! Just started seeing notifications for this thread on the way to work. I happened to get this running on Windows 10 thanks to this and another thread, with some kinks in the upload process. (It sounds like a change to the upload.js might solve my other issues.)

I've been meaning to write a Windows installation guide for a friend. If I get to that before this thread is answered, I'll post it here! (Though I think I'll be referring to this thread quite a bit to remember the steps. :) )

**Comment by ferguson4848** [comment]
That would be amazing. I'd really appreciate that.

Thanks

**Comment by AlBakerKY** [comment]
@ginareams There are quite a few updates that are needed AKAIK.  For example, the online simulation just doesn't work for me at all.  I can get Alexa to open the skill, but she will not accept any further commands after that.  I  did find that the test functionality of the skill on AWS itself was very useful for phonetic purposes. I did a puzzle game with this for Global Game Jam.

**Comment by ginareams** [comment]
@AlBakerKY When you say the online simulation, do you mean Echosim.io? I was able to test through the AWS test tool, Echoism.io, and my Alexa. So you can open your skill using "my skill" (or whatever prompt), but nothing else?

I've only got the tool running recently, but I've been able to run the example scenario (with the 4 doors) to the end on my device.

A puzzle game? That sounds neat! You mean using this tool or another?

**Comment by MureedSultan** [comment]
@ferguson4848 

- .aws folder needs to be created under C:\Users\YOUR_USERNAME\\.aws
- The folder needs to contain a file named 'credentials' (no extension) with the following contents:
```
[default]
aws_access_key_id = YOUR_ACCESS_KEY_ID
aws_secret_access_key = YOUR_SECRET_ACCESS_KEY
```

You can also set windows env variables AWS_ACCESS_KEY_ID & AWS_SECRET_ACCESS_KEY.
Refer to the Configuration Settings and Precedence section [here](http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html).

**Comment by AlBakerKY** [comment]
@ginareams Correct.  It would load the initial scene, but no interaction beyond that.  I tried it in chrome and edge.  

Yes, using this tool. It is game that lets you, conceptually, mask sentences to get new meaning from them. I wanted to see if I could come up with a game mechanic other than just exploring. It worked out pretty well. I didn't use any JS, but I would like to explore figuring out how to hook into the code so I could set session variables and hit Dynamo on my own.

**Comment by ginareams** [comment]
@AlBakerKY Were you able to test your Puzzle Game through Device though or submit it?

**Comment by AlBakerKY** [comment]
@ginareams Yes.  I could use it directly through Alexa with virtually no delay. The editor works ok, but if I had to do it again, I would probably adopt a more advanced JSON editor until they have cut/paste and move functionality. I am not sure if I will submit yet.

**Comment by nimzoking** [comment]
Coming to this a little late!  However, on Windows-10 I created a Linux VM with Hyper-V and do all the npm / github stuff on the Linux VM.  Works a treat, and I do not need to worry about Windows trying to emulate Linux :)

For an editor, I use Notepad++, it runs perfectly fine in Windows-10.

**Comment by Whitwood** [comment]
Was a Windows installation guide ever produced for this ?    I could really do with some help

**Comment by Manpuncher** [comment]
Yeah, I'm up to a point where I've got Node and NPM installed, I've created the .aws directory with the credentials file inside, but I appear unable to start npm the way it expects to be run.
